docs/configuration-files.md

```diff
@@ -272,91 +276,90 @@ rules:
 
 This rule looks for usage of Django's [`FloatField`](https://docs.djangoproject.com/en/3.0/ref/models/fields/#django.db.models.FloatField) model when storing currency information. `FloatField` can lead to rounding errors and should be avoided in favor of [`DecimalField`](https://docs.djangoproject.com/en/3.0/ref/models/fields/#django.db.models.DecimalField) when dealing with currency. Here the `pattern-where-python` operator allows us to utilize the Python `in` statement to filter findings that look like currency.
 
-## Metavariable matching
+## Metavariable Matching
 
-### Metavariables in logical inclusions
+Metavariable matching operates differently for logical AND (`patterns`)
+and logical OR (`pattern-either`) parent operators. Note that the behavior is
+consistent across all child operators: `pattern`, `pattern-not`,
+`pattern-regex`, `pattern-inside`, `pattern-not-inside`.
 
-Patterns' matched metavariable values are enforced to be identical when performing logical inclusion operations (`patterns`, `pattern-inside`) on matches.
+### Metavariables in Logical ANDs
+
+Metavariable values must be identical across sub-patterns when performing
+logical AND operations with the `patterns` operator.
```

Comment:
This should mention `pattern-inside`, as all the real-world examples would use `pattern-inside`.

Reply:
What's special about `pattern-inside` vs. say `pattern-not-inside`? My understanding is that all child operators have consistent behavior under the `patterns` operator.

When trying to break this down I see it as: metavariables must match under the `patterns` operator, and don't need to match under the `pattern-either` operator. From there, all child operators essentially act as you would expect.

So instead of thinking about this in 3 special cases: 1.) (`patterns`, `pattern-inside`), 2.) (`patterns`, `pattern-not-inside`), 3.) (`pattern-either`, `pattern`), I see it as (`patterns`, `<any-child-operator>`) and (`patterns-either`, `<any-child-operator>`).

Am I missing something important about the `inside` operators?

Reply:
See above re. behavior of `-inside`. I might be (very) confused here, but IIRC, `pattern-inside` does not conjugate like an `AND`.

Reply:
TBH, I haven't thought through what happens with metavariables when `pattern` comes before `pattern-inside` inside a `patterns` block...